contrib/node/src/python/pants/contrib/node/tasks/apache_thrift_node_gen.py

```diff
@@ -0,0 +1,56 @@
+import os
+import json
+from pants.backend.codegen.thrift.lib.apache_thrift_gen_base import ApacheThriftGenBase
+from pants.util.dirutil import safe_file_dump
+from pants.contrib.node.targets.node_thrift_library import NodeThriftLibrary
+from pants.contrib.node.targets.node_module import NodeModule
+
+
+class ApacheThriftNodeGen(ApacheThriftGenBase):
+  """Generate Nodejs source files from thrift IDL files"""
+  gentarget_type = NodeThriftLibrary
+  thrift_generator = 'js:node'
+  gen_directory = 'nodejs'
+
+  _COMPILER = 'thrift'
+
+  sources_globs = ('**/*',)
+
+  def synthetic_target_type(self, target):
```

Comment:
I think this would be the type hint?

```python
def synthetic_target_type(self, target: Target) -> NodeModule:
  ...
```

Where `Target` is imported at the top of this file from `pants.build_graph.target`.